tests/rest/test_connection.py

```diff
@@ -513,6 +513,86 @@ def test_capabilities_caching(requests_mock):
     assert con.capabilities().api_version() == "1.0.0"
     assert m.call_count == 1
 
+def test_capabilities_caching_after_authenticate_basic(requests_mock):
+    user, pwd = "john262", "J0hndo3"
+
+    def get_capabilities(request, context):
+        endpoints = BASIC_ENDPOINTS
+        if "Authorization" in request.headers:
+            endpoints.append({"path": "/account/status", "methods": ["GET"]})
+        return {"api_version": "1.0.0", "endpoints": endpoints}
+
+    get_capabilities_mock = requests_mock.get(API_URL, json=get_capabilities)
+    requests_mock.get(API_URL + 'credentials/basic', text=_credentials_basic_handler(user, pwd))
+
+    con = Connection(API_URL)
+    assert con.capabilities().capabilities == {
+        "api_version": "1.0.0",
+        "endpoints": [
+            {"methods": ["GET"], "path": "/credentials/basic"},
+        ],
+    }
+    assert get_capabilities_mock.call_count == 1
+    con.capabilities()
+    assert get_capabilities_mock.call_count == 1
+
+    con.authenticate_basic(user, pwd)
+    assert get_capabilities_mock.call_count == 1
+    assert con.capabilities().capabilities == {
+        "api_version": "1.0.0",
+        "endpoints": [
+            {"methods": ["GET"], "path": "/credentials/basic"},
+            {"methods": ["GET"], "path": "/account/status"},
+        ],
+    }
+    assert get_capabilities_mock.call_count == 2
+
+
+
+def test_capabilities_caching_after_authenticate_oidc(requests_mock):
+    requests_mock.get(API_URL, json={"api_version": "1.0.0"})
```

Comment:
this requests_mock.get call is unnecessary (is overwritten a couple of lines lower)